src/deadline/client/cli/_groups/asset_group.py

```diff
@@ -33,11 +40,61 @@ def cli_asset():
     default=False,
 )
 @_handle_error
-def asset_snapshot(**args):
+def asset_snapshot(root_dir: str, manifest_out: str, recursive: bool, **args):
     """
     Creates manifest of files specified root directory.
     """
-    click.echo("snapshot taken")
+    if not os.path.isdir(root_dir):
+        raise NonValidInputError(f"Specified root directory {root_dir} does not exist. ")
+
+    if manifest_out and not os.path.isdir(manifest_out):
+        raise NonValidInputError(f"Specified destination directory {manifest_out} does not exist. ")
+    elif manifest_out is None:
+        manifest_out = root_dir
+
+    inputs = []
+    for root, dirs, files in os.walk(root_dir):
+        inputs.extend([str(os.path.join(root, file)) for file in files])
+        if not recursive:
+            break
+
+    # Placeholder Asset Manager
+    asset_manager = S3AssetManager(
+        farm_id=" ", queue_id=" ", job_attachment_settings=JobAttachmentS3Settings(" ", " ")
+    )
+    asset_uploader = S3AssetUploader()
+    hash_callback_manager = _ProgressBarCallbackManager(length=100, label="Hashing Attachments")
+
+    upload_group = asset_manager.prepare_paths_for_upload(
+        input_paths=inputs, output_paths=[root_dir], referenced_paths=[]
```

Comment:
Curious - what is this `referenced_paths`?

Reply:
There's some comments on this page: https://github.com/ddneilson/deadline-cloud-samples/blob/the_job_bundle/job_bundles/README.md

tldr: it's neither input nor output, but in case job references the name of the folder; in the case of snapshot + upload, the main goal is to upload files we care about, so I don't think referenced_paths are relevant here.